### DISCUSSION TITLE
DWGADD How-To

### DISCUSSION THREAD

**Discussion by 64sys** [LibreDWG/libredwg#344] [Q&A]
Hi.
Here I summarize what I have so far managed to get working and understand regarding the file needed for dwgadd.
1 ) The end of the carriage is "\r\n" and not "\n". Right?
2) There has to be at least one # header. Right?
3) Add a Line in this way  **line (1 2 0) (2 3 0)** worked but this did not work 
line (1 2 0) (2 3 0) 
line.color = 3 (nor line.color = "3") // How to define a line color?
line.layer = "name"  // How to define a line layer?
4) The **dwgadd.example** and the command **dwgadd -o tmp.dwg dwgadd.example**
Creates a 21 kb file called **tmp.dwg** but is necessary recover it in autocad in order to open it. next  is possible open it, this is great, but how to add properly a line ? Then how to change the colors, layers and others properties for the new entities added?? 
Am I doing something wrong? 

Does anyone have an example of this accessory file for dwgadd where the added entities have colors and layers?
Regards.

**Comment by 64sys** [comment]
Hello again, even if it is an example with a line and then I continue.
It is that at the moment the dwg created is all white, without layers and without styles. also it is necessary to restore it.
If someone has an example it would be good for me.
Thanks

**Comment by omar-khan-post** [comment]
I have been trying myself to the point of looking and recompiling the c code  @belmotek 
The problem is dwgadd.c file only modifies the entity not the **parent object.**
See bool dwg_dynapi_entity_set_value vs [dwg_dynapi_common_set_value.](https://www.gnu.org/software/libredwg/manual/html_node/dynapi.html#dynapi)
Layer and color are part of the common parent set not the entity set.

> The new add API mostly handles the direct Dwg_Entity_ENTITY structs, not all the generic Dwg_Object structs. Thus you can access the object specific fields directly, the common fields, not so easily.




You would need to modify #define SET_ENT(var, name)   I believe.  But if you gotten this far, might as well write c code.

**Reply by yingyue4008** [reply]
I also encountered the above problems. For example, after creating a new layer, I replaced the old layer on the original line with the new layer. I assigned the layer xref, but opened a wrong layer instead of a new one in dwg, so I don't know what to do, so I asked for help
